tests/components/mailbox/test_init.py

```diff
@@ -0,0 +1,158 @@
+"""The tests for the mailbox component."""
+import json
+import requests
+from hashlib import sha1
+
+from homeassistant.setup import setup_component
+import homeassistant.components.mailbox as mailbox
+import homeassistant.components.http as http
+
+from tests.common import (
+    get_test_home_assistant, get_test_instance_port, assert_setup_component)
+
+
+class TestSetupMailbox(object):
```

Comment:
You can rewrite this whole test class as a test function on the root level. Use `hass` as a parameter to the test function to automatically get a new test instance:

```python
def test_setup_component(has):
    """Initialize demo platform on mailbox component."""
    config = {
        mailbox.DOMAIN: {
            'platform': 'demo'
        }
    }

    with assert_setup_component(1, mailbox.DOMAIN):
        setup_component(hass, mailbox.DOMAIN, config) 
```